client/src/main/java/io/pravega/client/batch/impl/SegmentIteratorImpl.java

```diff
@@ -95,4 +108,47 @@ public void close() {
         input.close();
     }
 
+    static class SegmentIteratorBuilder<T> implements ObjectBuilder<SegmentIteratorImpl> {
+    }
+
+    private static class SegmentIteratorSerializer extends VersionedSerializer.WithBuilder<SegmentIteratorImpl, SegmentIteratorBuilder<Object>> {
+        @Override
+        protected byte getWriteVersion() {
+            return 0;
+        }
+
+        @Override
+        protected void declareVersions() {
+            version(0).revision(0, this::write00, this::read00);
+        }
+
+        @Override
+        protected SegmentIteratorBuilder<Object> newBuilder() {
+            return builder();
+        }
+
+        private void read00(RevisionDataInput revisionDataInput, SegmentIteratorImpl.SegmentIteratorBuilder builder) throws IOException {
+            builder.segment(Segment.fromScopedName(revisionDataInput.readUTF()));
+            builder.startingOffset(revisionDataInput.readCompactLong());
+            builder.endingOffset(revisionDataInput.readCompactLong());
+            builder.factory(factory);
+        }
+
+        private void write00(SegmentIteratorImpl segmentIterator, RevisionDataOutput revisionDataOutput) throws IOException {
+            revisionDataOutput.writeUTF(segmentIterator.segment.getScopedName());
+            revisionDataOutput.writeCompactLong(segmentIterator.getStartingOffset());
+            revisionDataOutput.writeCompactLong(segmentIterator.endingOffset);
+        }
+    }
+    @Override
+    @SneakyThrows(IOException.class)
+    public ByteBuffer toBytes() {
+        ByteArraySegment serialized = SERIALIZER.serialize(this);
+        return ByteBuffer.wrap(serialized.array(), serialized.arrayOffset(), serialized.getLength());
+    }
+    @Override
+    @SneakyThrows(IOException.class)
+    public  SegmentIterator fromBytes(ByteBuffer serializedPosition) {
+        return SERIALIZER.deserialize(new ByteArraySegment(serializedPosition));
```

Comment:
This is not going to set `deserializer` which will cause a null pointer exception.

Reply:
this is also taken care.